src/main/kotlin/com/openlattice/graph/Graph.kt

```diff
@@ -1186,6 +1190,10 @@ private val EDGE_IDS_SQL = "${EDGE_ENTITY_KEY_ID.name} = ANY(?) AND ${EDGE_ENTIT
 private val SRC_IDS_AND_ENTITY_SETS_SQL = "${SRC_ENTITY_KEY_ID.name} = ANY(?) AND ${SRC_ENTITY_SET_ID.name} = ANY(?)"
 private val DST_IDS_AND_ENTITY_SETS_SQL = "${DST_ENTITY_KEY_ID.name} = ANY(?) AND ${DST_ENTITY_SET_ID.name} = ANY(?)"
 
+const val REPARTITION_SELECTOR = "partitions[ 1 + (('x'||right(id::text,8))::bit(32)::int % array_length(partitions,1))]"
```

Comment:
if I remember right, there needs to be an `abs()` somewhere in here for when ids end up wrapping (/any leftover ones we had from when id generation was just random uuid)

Reply:
We shouldn't have any, but I have a fix for this issue which will match what the back-end does. This code merge is more fore hz-4.0 and I'm going to keep working on this branch.